hypershift-operator/controllers/nodepool/kubevirt/kubevirt.go

```diff
@@ -147,64 +150,48 @@ func virtualMachineTemplateBase(image string, kvPlatform *hyperv1.KubevirtNodePo
 		},
 	}
 
+	if kvPlatform.RootVolume != nil {
+		if kvPlatform.RootVolume.Persistent != nil && kvPlatform.RootVolume.Persistent.Size != nil {
+			volumeSize = *kvPlatform.RootVolume.Persistent.Size
+		}
+
+		if kvPlatform.RootVolume.Persistent != nil {
+			storageClass = kvPlatform.RootVolume.Persistent.StorageClass
+
+			for _, am := range kvPlatform.RootVolume.Persistent.AccessModes {
+				accessModes = append(accessModes, corev1.PersistentVolumeAccessMode(am))
+			}
+		}
+	}
+
 	dataVolume := kubevirtv1.DataVolumeTemplateSpec{
 		TypeMeta: metav1.TypeMeta{},
 		ObjectMeta: metav1.ObjectMeta{
 			Name: rootVolumeName,
 		},
-	}
-
-	if strings.HasPrefix(image, "docker://") {
-		dataVolume.Spec = v1beta1.DataVolumeSpec{
-			Source: &v1beta1.DataVolumeSource{
-				Registry: &v1beta1.DataVolumeSourceRegistry{
-					URL:        &image,
-					PullMethod: &pullMethod,
-				},
-			},
-		}
-	} else {
-		dataVolume.Spec = v1beta1.DataVolumeSpec{
-			Source: &v1beta1.DataVolumeSource{
-				HTTP: &v1beta1.DataVolumeSourceHTTP{
-					URL: image,
+		Spec: v1beta1.DataVolumeSpec{
+			Source: dvSource,
+			Storage: &v1beta1.StorageSpec{
+				AccessModes: accessModes,
```

Comment:
One of the points of using the `storage` api is that you don't have to specify the access mode, it is inferred from the `StorageProfile` which should contain the optimal accessMode and volumeMode for your storage.

Reply:
> As I read the code it seems to me the cache is doing the exact same function as DataImportCrons minus the automatic updates on a cron schedule.

This is very similar, indeed. But again, this is for qcow images. These images are not supported by the DataImportCron.

Reply:
> One of the points of using the storage api is that you don't have to specify the access mode, it is inferred from the StorageProfile which should contain the optimal accessMode and volumeMode for your storage.

This part still needs to be addressed.